src/main/java/com/mapzen/valhalla/HttpHandler.java

```diff
@@ -0,0 +1,65 @@
+package com.mapzen.valhalla;
+
+import com.mapzen.helpers.ResultConverter;
+import com.mapzen.valhalla.RestAdapterFactory;
+import com.mapzen.valhalla.RoutingService;
+
+import retrofit.Callback;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+
+/**
+ *
+ */
+public class HttpHandler {
+
+  private static final String DEFAULT_URL = "https://valhalla.mapzen.com/";
+  private static final RestAdapter.LogLevel DEFAULT_LOG_LEVEL = RestAdapter.LogLevel.NONE;
+
+  String apiKey;
+  String endpoint;
+  RestAdapter.LogLevel logLevel;
+  RestAdapter adapter;
+  RoutingService service;
+
+  private RequestInterceptor requestInterceptor = new RequestInterceptor() {
+    @Override public void intercept(RequestFacade request) {
+      addHeadersForRequest(request);
+    }
+  };
+
+  public HttpHandler(String apiKey) {
+    this(apiKey, DEFAULT_URL, DEFAULT_LOG_LEVEL);
```

Comment:
I'm thinking we should completely remove the concept of an API key from this project since it is specific to the Mapzen instance of Valhalla.

Similar to `addHeadersForRequest(...)`, the default `HttpHandler` defined here should just make it possible to append arbitrary query params to outgoing requests.

Then the Mapzen Android SDK can extend this class to append the API key param, similar to the [`TileHttpHandler`](https://github.com/mapzen/android/blob/master/library/src/main/java/com/mapzen/android/TileHttpHandler.java).

What do you think?